File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String a, String b) {
        if(a.length()!=b.length())return false;
        Map<Character, Integer> map = new HashMap<>();

        for (Character c : a.toCharArray()) {
            if (map.containsKey(c))
                map.put(c, map.get(c) + 1);
            else
                map.put(c, 1);
        }
        for (Character c : b.toCharArray()) {
            if (map.containsKey(c)) {
                if (map.get(c) == 0)
                    return false;
                else
                    map.put(c, map.get(c) - 1);
            } else
                return false;
        }

        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> map;
        for(char c : s) map[c]++;
        
        for(char c : t) {
            if(map[c] <= 0) return false;
            map[c]--;
        }
        
        for(auto it : map) {
            if(it.second > 0) return false;
        }
        return true;
    }
};